src/foundry/client/pixi/placeables/primary-canvas-objects/primary-canvas-object.d.ts

```diff
@@ -0,0 +1,148 @@
+   */
+  static defaultData: PrimaryCanvasObject.Data;
+
+  /**
+   * An elevation in distance units which defines how this Object is sorted relative to its siblings.
+   */
+  get elevation(): number;
+
+  /**
+   * A sort key which resolves ties amongst objects at the same elevation.
+   */
+  get sort(): number;
+
+  /**
+   * A convenient reference to a Document associated with this display object, if any.
+   */
+  get document(): ClientDocument | null;
+
+  /**
+   * Initialize data using an explicitly provided data object or a canvas document.
+   * @param data - Provided data or canvas document.
+   */
+  initialize(data?: PrimaryCanvasObject.Data | Document): void;
+
+  /**
+   * Map the document data to an object and process some properties.
+   * @param data - The document data.
+   * @returns The updated data object.
+   */
+  protected _getCanvasDocumentData(data: Document): unknown;
+
+  /**
+   * Initialize sorting of this PCO. Perform checks and call the primary group sorting if necessary.
+   * @param sort - The sort value. Must be a finite number or undefined (in this case, it is ignored)
+   */
+  protected _initializeSorting(sort: number): void;
+
+  destroy(...args: any[]): void;
+
+  /**
+   * Synchronize the appearance of this ObjectMesh with the properties of its represented Document.
+   * @remarks Marked as abstract
+   */
+  refresh(): void;
+
+  /**
+   * Synchronize the position of the ObjectMesh using the position of its represented Document.
+   * @remarks Marked as abstract
+   */
+  setPosition(...args: any[]): void;
+
+  /**
+   * Synchronize the bounds of the ObjectMesh into the primary group quadtree.
+   */
+  updateBounds(): void;
+}
+
+declare global {
+  namespace PrimaryCanvasObject {
+    type Data = {
+      /** The x-coordinate of the PCO location */
+      x: number;
+
+      /** The y-coordinate of the PCO location */
+      y: number;
+
+      /** The z-index of the PCO */
+      z: number;
+
+      /** The width of the PCO */
+      width: number;
+
+      /** The height of the PCO */
+      height: number;
+
+      /** The alpha of this PCO */
+      alpha: number;
+
+      /** The rotation of this PCO */
+      rotation: number;
+
+      /** The PCO is hidden? */
+      hidden: boolean;
+
+      /** The elevation of the PCO */
+      elevation: number;
+
+      /** The sort key that resolves ties among the same elevation */
+      sort: number;
+
+      /** The data texture values */
+      texture: PIXI.RenderTexture;
+    };
+  }
+  /**
+   * A mixin which decorates a DisplayObject with additional properties expected for rendering in the PrimaryCanvasGroup.
+   * @param DisplayObject - The parent DisplayObject class being mixed
+   * @returns A DisplayObject subclass mixed with PrimaryCanvasObject features
+   */
+  function PrimaryCanvasObjectMixin<
+    BaseClass extends Pick<typeof PIXI.DisplayObject, keyof typeof PIXI.DisplayObject> &
+      (new (...args: any[]) => PIXI.DisplayObject),
```

Comment:
I think we added `AnyConstructor` for exactly this case. So you can do `BaseClass extends AnyConstructor<typeof PIXI.DisplayObject>`.

Reply:
So when I switch to
```ts
  function PrimaryCanvasObjectMixin<BaseClass extends AnyConstructor<typeof PIXI.DisplayObject>>(
    DisplayObject: BaseClass,
  ): Mixin<typeof PrimaryCanvasObject, BaseClass>;
```
I get "Type 'typeof DisplayObject' does not satisfy the constraint 'new (...args: any[]) => any'.
  Cannot assign an abstract constructor type to a non-abstract constructor type."
  
So the crux of the problem is that `DisplayObject` is `abstract`; any good way to purge that?

Reply:
Yeah I was just lazy when I wrote the initial type. I knew about the `abstract` case and didn't want to handle it because it makes the type look sillier.
```ts
type AnyConstructor<Class extends abstract new(...args: any[]) => any;> = Pick<Class, keyof Class> & (
    Class extends new (...args: any[]) => any ? new (...args: any[]) => InstanceType<Class> : abstract new(...args: any[]) => InstanceType<Class>
)
```

Ah well, here's the fixed type.